component/custom/custom_loader.go

```diff
@@ -0,0 +1,40 @@
+package custom
+
+import (
+	"os"
+	"plugin"
+	"strings"
+)
+
+func init() {
+	pluginPath := os.Getenv("PLUGIN_PATH")
+	if pluginPath == "" {
+		return
+	}
+
+	// environment variable can contain multiple paths separated by colon.
+	paths := strings.Split(pluginPath, ":")
+
+	for _, file := range paths {
+		println("custom_loader::Opening plugin:", file)
+		p, err := plugin.Open(file)
+		if err != nil {
+			println("custom_loader::Error opening plugin:", err.Error())
+			return
```

Comment:
Should you not throw ?

Reply:
was in doubt whether to fail a case of plugin issue. but yes i think it makes more sense to throw the error because config file will also expect this custom component.